exception-handling/raising-and-handling-exceptions.md

```diff
@@ -0,0 +1,306 @@
+
+## Introduction
+
+If you weren't sure, the absolute worst-case scenario for software is a crash caused by a syntax or runtime error\*. The software has gone to a state where the program can't do anything, the user can't do anything, and possibly the program can't even restart into a usable state!
+
+If software fails for one person, it's likely that software is failing for _many_ people. All of these consequences result in **chaos**.
+
+![giphy](https://user-images.githubusercontent.com/16619004/103056684-748d8a80-4552-11eb-8705-2429df508339.gif) Source: [Giphy](http://gph.is/2lENCEI)
+
+\* = Actually, there are plenty of other more severe worst-case scenarios.
+
+## Vocabulary and Synonyms
+
+| Vocab     | Definition                                                                     | Synonyms | How to Use in a Sentence                                                                      |
+| --------- | ------------------------------------------------------------------------------ | -------- | --------------------------------------------------------------------------------------------- |
+| Exception | An event or error that may cause your program to stop if not properly handled. | error    | "The user provided the wrong data type causing the Traceback to throw a TypeError exception." |
+
+## Exceptions Are Python's Cry For Help
+
+An exception is a type of _thing_ (we can say "data type") that holds information about an error that the program encounters. Exceptions hold data like specifically what error occured and which line of code caused it.
+
+Python creates an exception whenever a syntax or runtime error occurs in a program. Python will then immediately try to stop the execution of the program and exit out of it. When an error happens, and we see details in the stack trace, we actually see details about the exception!
+
+The act of creating an exception is often called **raising an exception.**
+
+When an exception is raised, Python moves "up" through the stack of code execution. This act of the exception moving "up" is often called **throwing an exception** or **bubbling up.** We might imagine an exception raised in a deep layer of code gets thrown up through the each layer of code, eventually to the programmer's face.
+
+Python raises exceptions because exceptions are Python's cry for help. Python only raises exceptions when some code says, "I can't do this! I can't go on!"
+
+Of course, when Python surrenders and throws an exception, we can't force it to keep going. Ideally, however, should we see an exception, we'd love to recognize it, handle the situation, and then allow our program to move on.
+
+Code that recognizes a thrown exception is often called **catching an exception.** Catching an exception and then handling the situation, setting things back in order, and making it possible to move on, is often called **handling an exception** or **rescuing an exception.**
```

Comment:
Suggest avoiding the ruby terminology.
```suggestion
Code that recognizes a thrown exception is often called **catching an exception.** Catching an exception and then handling the situation, setting things back in order, and making it possible to move on, is also often called **handling an exception.**
```

Reply:
Didn't realize that was only a rubyism, but now that I think about it, I didn't learn "rescue" until Ruby haha. thanks for the catch!